lib/travis/build/appliances/services.rb

```diff
@@ -35,6 +31,25 @@ def apply?
           services.any?
         end
 
+        def apply_mongodb
+          sh.if "$(lsb_release -cs) != 'precise'" do
+            sh.cmd 'sudo service mongod start', assert: false, echo: true, timing: true
+          end
+          sh.else do
+            sh.cmd 'sudo service mongodb start', assert: false, echo: true, timing: true
+          end
+        end
+
+        def apply_mysql
+          sh.raw <<~BASH
+            travis_mysql_ping() {
+              mysql &>/dev/null <<<'SHOW VARIABLES like "%version%"'
+            }
+          BASH
+          sh.cmd 'sudo service mysql start', assert: false, echo: true, timing: true
+          sh.cmd 'travis_wait travis_mysql_ping', assert: false, echo: false, timing: false
```

Comment:
I am not sure if `travis_wait` is a good choice here. The smallest granularity that it gives is 1 minute, which might be too coerce. In addition, I am thinking about tweaking `travis_wait` a bit; have an idea, but have not worked on it.

A more straightforward thing to do is to check if the TCP port is open with some timeout.

```sh
until ruby -rsocket -e "TCPSocket.new('localhost',3306)" || test $count -lt 5; do sleep 1; let count=$count+1; done
```